agent_plugin_builder/agent_plugin_builder.py

```diff
@@ -1,66 +1,139 @@
+                default_str = default
+
+        if "%(default)" not in help_str and default is not None:
+            if action.dest != "help":
+                help_str += "(Default: %(default)s)"
+
+        return help_str % dict(default=default_str)
+
+
 def main():
-    parser = ArgumentParser(description="Build plugin")
-    parser.add_argument("plugin_path", metavar="PLUGIN_PATH", type=Path, help="Path to the plugin)")
+    parser = ArgumentParser(description="Build plugin", formatter_class=CustomFormatter)
+    parser.add_argument("plugin_path", metavar="PLUGIN_PATH", type=Path, help="Path to the plugin")
     parser.add_argument(
         "-b",
         "--build-dir-path",
         metavar="BUILD_DIR_PATH",
         type=Path,
         default=(Path.cwd() / BUILD),
-        help="Optional path to the build directory. Default: <current_working_directory>/build.",
+        help="Optional path to the build directory.\n",
     )
     parser.add_argument(
         "-d",
         "--dist-dir-path",
         metavar="DIST_DIR_PATH",
         type=Path,
         default=(Path.cwd() / DIST),
-        help="Optional path to the dist directory. Default: <current_working_directory>/dist.",
+        help="Optional path to the dist directory.\n",
     )
     parser.add_argument(
         "-s",
         "--source-dir",
         metavar="SOURCE_DIR",
         type=str,
         default=None,
-        help="Optional name of the source directory. Default: <plugin_name>_<plugin_type>.",
+        help="Optional name of the source directory.\n",
+    )
+    parser.add_argument(
+        "-pd",
+        "--platform-dependencies",
+        metavar="PLATFORM_DEPENDENCIES",
+        type=PlatformDependencyPackagingMethod,
+        default=PlatformDependencyPackagingMethod.SEPARATE,
+        help="""Since plugin dependencies must be vendored, this setting determines how
+the plugin builder should package dependencies for it's supported platforms.
+
+Options:
+    common: All dependencies are packaged once, and shared across all platforms. This
+            should only be used if all dependencies are known to be platform-independent.
+    separate: Some or all dependencies are platform-dependent, and are therefore packaged
+            separately for each supported platform. This is the most reliable option,
+            however it results in a larger plugin file, since dependencies are
+            duplicated for each platform.
+    autodetect: The plugin builder will attempt to detect the best method to use.
+""",
+    )
+    parser.add_argument(
+        "--hashes",
+        metavar="HASHES",
+        action=BooleanOptionalAction,
+        help="""Some plugins may have dependencies for which we can't verify the hashes.
+
+Options:
+    --hashes: will include hashes from the requirements.txt file
+    --no-hashes: will exclude hashes from the requirements.txt file which is not recommended
+    not specified: will include hashes from the requirements.txt file
```

Comment:
This is a low-level explanation of pip-specific feature. Whether or not hashes are included in the requirements.txt file is meaningless to the user, especially since the user doesn't even know a requirements.txt file is generated. That's an internal implementation detail of the tool, and the whole point of the tool is to insulate users from such details.

Reply:
Should I not explain this or should I just rewrite the description of what this meant ?